exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillResultSetImpl.java

```diff
@@ -125,7 +154,7 @@ protected void cancel() {
   // (Not delegated.)
   @Override
   public boolean next() throws SQLException {
-    throwIfClosed();
+    throwIfTimedOutOrClosed();
```

Comment:
if we chose to support `queryTimeout` for `ResultSet`, shouldn't we interrupt `next()` too if the operation is taking too long? As per your code, it seems the exception would be thrown after facts...